wordpress-mobile/WordPressAuthenticator-iOS
# Log In: show login options after site address validated.

This change allows displaying a login options button view after a site address has been validated. As this change is specifically for Woo, it should be tested with the Woo app.

(Related discussion can be found [here](https://wp.me/pauD4L-li-p2/#comment-268).)

**Disclaimers**: 
- Woo has not adopted the latest WPAuth pod, and there are several colors `WordPressAuthenticatorStyle` now requires. I picked some to get it working, but they're obviously incorrect. Please ignore the colors in these examples as this PR is specific to the flow, and I assume those colors will be set correctly when Woo updates WPAuth.
- This change assumes in the future Woo will start the login process from the site address view. So there have been no changes to the initial login view (i.e. the email entry).

---
**Prerequisite** - required to get Woo working with the latest WPAuth pod:
- In `AuthenticationManager`, add the missing `WordPressAuthenticatorStyle` parameters. If it helps, you can simply replace the entire init with this, which includes my stub colors.

```
let style = WordPressAuthenticatorStyle(primaryNormalBackgroundColor: StyleManager.buttonPrimaryColor,
                                        primaryNormalBorderColor: StyleManager.buttonPrimaryHighlightedColor,
                                        primaryHighlightBackgroundColor: StyleManager.buttonPrimaryHighlightedColor,
                                        primaryHighlightBorderColor: StyleManager.buttonPrimaryHighlightedColor,
                                        secondaryNormalBackgroundColor: StyleManager.buttonSecondaryColor,
                                        secondaryNormalBorderColor: StyleManager.buttonSecondaryHighlightedColor,
                                        secondaryHighlightBackgroundColor: StyleManager.buttonSecondaryHighlightedColor,
                                        secondaryHighlightBorderColor: StyleManager.buttonSecondaryHighlightedColor,
                                        disabledBackgroundColor: StyleManager.buttonDisabledColor,
                                        disabledBorderColor: StyleManager.buttonDisabledHighlightedColor,
                                        primaryTitleColor: StyleManager.buttonPrimaryTitleColor,
                                        secondaryTitleColor: StyleManager.buttonSecondaryTitleColor,
                                        disabledTitleColor: StyleManager.buttonDisabledTitleColor,
                                        textButtonColor: StyleManager.buttonPrimaryColor,
                                        textButtonHighlightColor: StyleManager.buttonPrimaryHighlightedColor,
                                        instructionColor: StyleManager.defaultTextColor,
                                        subheadlineColor: StyleManager.wooCommerceBrandColor,
                                        placeholderColor: StyleManager.defaultTextColor,
                                        viewControllerBackgroundColor: StyleManager.wooGreyLight,
                                        navBarImage: StyleManager.navBarImage,
                                        navBarBadgeColor: StyleManager.defaultTextColor,
                                        prologueBackgroundColor: StyleManager.wooCommerceBrandColor,
                                        prologueTitleColor: StyleManager.wooSecondary
                                        )
```                                                

---
**Prepare for testing** - in the Woo app:
- Change `WordPressAuthenticator` pod to use this branch (`feature/new_login_flow_site_address`).
- In `AuthenticationManager`, add `showLoginOptionsFromSiteAddress: true` as the last parameter to the `WordPressAuthenticatorConfiguration` init.

---
**Testing**:
- Run the app. Log out if necessary.
- Select 'Log in with Jetpack'. 
- Select 'Log in by entering your site address.'
- Enter a site address and select 'Next'. After the site is validated:
  - If it's a self-hosted site, the existing self-hosted flow is displayed.
  - If it's a WP site, the new options are shown.

---
**For WP sites**:
- A button view is displayed with login options. (Again, please ignore the colors.)

![login_options](https://user-images.githubusercontent.com/1816888/63067230-2b8fe580-becb-11e9-87e0-2fb802143db6.png)

- 'Continue with WordPress.com' launches the existing WP login flow.

![wp_login](https://user-images.githubusercontent.com/1816888/63067253-3e0a1f00-becb-11e9-9787-6e3d9d345d98.png)

- 'Continue with Google' launches the Google authentication flow.

![google](https://user-images.githubusercontent.com/1816888/63067081-7e1cd200-beca-11e9-8a41-b54aedc4b23f.png)

---
cc @astralbodies , @frosty
